pyvista/core/filters/data_set.py

```diff
@@ -3161,6 +3208,9 @@ def sample(
         if tolerance is not None:
             alg.SetComputeTolerance(False)
             alg.SetTolerance(tolerance)
+        if locator:
+            alg.SetCellLocatorPrototype(locator)
```

Comment:
I think this is nice to expose this interface for "power users" but should we consider supporting named locators? For example, allowing users to pass the string name of a locator type and automatically generating it for them?

Reply:
This was done to make it a like-for-like for existing `probe` filter.  We could in the future make it so the user can add string instead and then we give them the default instantiated locator.  I'll that to the enhancement list in #4658.